book/integration-service.md

```diff
@@ -0,0 +1,271 @@
+- [Enterprise Contract Service](./enterprise-contract.md)
+  - Provides facilities to validate whether content has passed the Enterprise Contract.
+
+
+## Terminology
+
+**Prelim Pipeline** - This is the test pipeline run for the Component of an Application that gets triggered by a completed Build pipeline. Deprecated name, refer to the Component Pipeline
+
+**Component Pipeline** - This is the same as the Prelim pipeline. Going forward the Prelim Pipeline will be referred to as Component Pipeline
+
+**Final Pipeline** - This is the pipeline run that runs if two or more Components have completed and passed their Component pipeline concurrently so that they should be tested together. Deprecated name, refer to Composite Pipeline.
+
+**Composite Pipeline** - This is the same as the Final Pipeline. Going forward the Final Pipeline will be referred to as Composite Pipeline
+
+**Global Candidate List** - The list of all Component digest that have passed individual component testing. This can be retrieved from the Application to see what Components it is made of and then querying each of the Components `spec.containerImage`. 
+
+**Snapshot** - The list that maps all Components of an Application to their Component Image digests. Once created, the snapshot is immutable. 
+
+**Snasphot Digest** - The snapshot itself should have an identifier constructed from its contents - using a digest of the contents likely makes sense.
+
+**Integration Service** - The service that uses the pipeline, snapshot and environment controllers to watch for Component builds, trigger and manage the testing pipelines and create releases based on the testing outcome(s).
+
+**IntegrationTestScenario** - The custom resource that describes separate PipelineRuns that are supposed to be run by the snapshot controller. Any number of them can be set by the user and each of them includes information about the Tekton pipeline that will be executed - Tekton bundle, pipeline name and parameters, and the Environment details. The Integration Test Scenarios can also specify the contexts where they will be applied - this can be used to define when the scenario will run (Component or Composite stage - or both). If more than one Integration Test Scenario has the same context, their respective PipelineRuns will be executed in parallel.
+
+**Ephemeral Environment** - The ephemeral copy of an existing environment that’s created by the Integration service for use during a specific Integration pipeline run. After the environment is copied and marked as ephemeral, the Integration service creates a new SnapshotEnvironmentBinding linking the Snapshot under test with the newly created Environment. This triggers a deployment of the Snapshot on the new Environment, handled by the GitOps service. Finally, the Integration service links the credentials for the Environment to the Integration Tekton pipeline run and starts the test run.
```

Comment:
The snapshot is just defined as a list above. We are not deploying a list to the environments but an immutable set of components.

Reply:
I think that you have included important details in this bullet, but I feel like most of it would be better to include elsewhere -- potentially in the steps carried out by integration service. This should be a definition of an ephemeral environment not a description of what actions are being applied to the environment after creation. If too much detail for the overview, would some place like the generalized workflow be better?

Reply:
Done